File: WebClient/src/app/models/basic-item.ts
export class BasicItem {
  listId: number;
  storeId: number;
  name: string;
  price: number;
  salePrice: number;
  quantity: number;
  availFrom: string; // CHANGE TO DATE
  availTo: string; // CHANGE TO DATE
  blob: string;
  categoryId: number;

  constructor(
    listId: number,
    storeId: number,
    name: string,
    price: number,
    salePrice: number,
    quantity: number,
    availFrom: string,
    availTo: string,
    blob: string,
    categoryId: number
  )
  {
    this.listId = listId;
    this.storeId = storeId;
    this.name = name;
    this.price = price;
    this.salePrice = salePrice;
    this.quantity = quantity;
    this.availFrom = availFrom;
    this.availTo = availTo;
    this.blob = blob;
    this.categoryId = categoryId;
  }
}

File: Server/Models/BasicItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System;


namespace Server.Models
{
    public class BasicItem{
        [Key]
        [Column("list_id")]
        public int ListId {get; set;}
        [Column("store_id")]
        public int StoreId {get; set;}
        [Column("name")]
        public string Name {get; set;} = string.Empty;
        [Column("price")]
        public decimal Price {get; set;}
        [Column("sale_price")]
        public decimal SalePrice {get; set;}
        [Column("quantity")]
        public int quantity {get; set;}
         [Column("availFrom")]
        public DateTime? AvailFrom {get; set;}
         [Column("availTo")]
        public DateTime? AvailTo {get; set;}
         [Column("blob")]
        public string? Blob {get; set;}
        [Column("category")]
        public int CategoryId {get; set;}
    }
}